lib/Support/TimingModels.cpp

```diff
@@ -62,12 +62,13 @@ unsigned dynamatic::getOpDatawidth(Operation *op) {
       .Case<handshake::SourceOp, handshake::ConstantOp>([&](auto) {
         return getHandshakeTypeBitWidth(op->getResult(0).getType());
       })
-      .Case<handshake::EndOp, handshake::JoinOp>([&](auto) {
-        unsigned maxWidth = 0;
-        for (Type ty : op->getOperandTypes())
-          maxWidth = std::max(maxWidth, getHandshakeTypeBitWidth(ty));
-        return maxWidth;
-      })
+      .Case<handshake::EndOp, handshake::JoinOp, handshake::BlockerOp>(
```

Comment:
Since we are enforcing `SameOperandsAndResultType`, we don't need to calculate the `max` of all operand bitwidths.
It actually also applies to `JoinOp` though...

Reply:
@DanaKossaybati actually `EndOp` needs `max`, so I think you need to keep this case.
How about ading `JoinOp` and `BlockerOp` to the cases for `BufferOp`, `ForkOp`, ...?

But general question: Do you need `BlockerOp` handling here? I don't see a timing model added for it. If it's not required, you can revert this file to its original state.